File: back-end/src/main/java/com/trucks/application/service/PortService.java
package com.trucks.application.service;

import java.util.List;
import java.util.UUID;
import java.util.Optional;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.server.ResponseStatusException;

import com.trucks.application.port.PortRepository;
import com.trucks.domain.entity.Port;

public class PortService {

    @Autowired
    PortRepository portRepository;

    public Page<Port> findAll(PageRequest pageRequest) {
        return this.portRepository.findAll(pageRequest);
    }

    public Port save(Port newPort) {
        return this.portRepository.save(newPort);
    }

    public Optional<Port> find(UUID id) {
        return this.portRepository.find(id);
    }

    public Port delete(UUID id) {
        return this.portRepository.delete(id);
    }

    public Port update(UUID id, Port updatedPort) {
        return this.portRepository.update(id, updatedPort);
    }

    public List<Port> deleteAll() {
        return this.portRepository.deleteAll();
    }
}

File: front-end/src/application/service/portService.ts
import { PortRepository } from "../repository/portRepository";
import { Port } from "../../domain/entity/port";
import { Either } from "../../utils/ull-either";
import { DataError } from "../../utils/ull-data-error";
import { UllUUID } from "../../utils/ull-uuid";

export class PortService {
    private portRepository: PortRepository;

    constructor(portRepository: PortRepository) {
        this.portRepository = portRepository;
    }
    
    public save = (newPort: Port): Promise<Either<DataError, Port>> => {
        return this.portRepository.save(newPort);
    }
    
    public find = (id: UllUUID): Promise<Either<DataError, Port>> => {
        return this.portRepository.find(id);
    }
    
    public findAll = (pageRequest: any): Promise<Either<DataError, Port[]>> => {
        return this.portRepository.findAll(pageRequest);
    }

    public update = (id: UllUUID, updatedPort: Port): Promise<
        Either<DataError, Port>> => {
            return this.portRepository.update(id, updatedPort);
    }
    
    public delete = (id: UllUUID): Promise<Either<DataError, Port>> => {
        return this.portRepository.delete(id);
    }

    public deleteAll = (): Promise<Either<DataError, Port[]>> => {
        return this.portRepository.deleteAll();
    }
}